### PR TITLE
Added caution-box about special characters

### PR DESCRIPTION
**Pull request by ThomasLandauer**
Copied caution-box about special characters from https://symfony.com/doc/current/doctrine.html#configuring-the-database

### CODE REVIEW

File: email.rst [hunk 1]
```diff
@@ -39,11 +39,16 @@ environment variable in the ``.env`` file:
...
-    # values of the username and password if they contain non-alphanumeric characters
-    # such as '+', '@', ':' and '*', which are reserved in URLs)
+    # use this to configure a traditional SMTP server
     MAILER_URL=smtp://localhost:25?encryption=ssl&auth_mode=login&username=&password=
 
+.. caution::
+
+    If the username, password or host contain any character considered special in a
+    URI (such as ``+``, ``@``, ``$``, ``#``, ``/``, ``:``, ``*``, ``!``), you must
+    encode them. See `RFC 3986`_ for the full list of reserved characters or use the
```
Comment (OskarStark) [inline email.rst]: please copy the link, too!

```
Warning, treated as error:
/home/travis/build/symfony/symfony-docs/email.rst:47: ERROR: Unknown target name: "rfc 3986".
```

**Review by OskarStark — CHANGES_REQUESTED** [review]

**Comment by OskarStark** [comment]
shall we move it to an include instead of duplicating it? or is it not exactly the same content?

**Comment by ThomasLandauer** [comment]
Include: I dont't know how to do that.

Differences:
* I deleted the term "database". I could come up with another PR at Doctrine, using a more general term like "credentials" or anything.
* And I added some more example characters. These are not so important, and I could add them at Doctrine too.

So if you go for the include, I'll take care of the fine tuning :-)

BTW: I'm not 100% sure if Swift Mailer does indeed use the same decoding as Doctrine, see https://github.com/symfony/swiftmailer-bundle/issues/270 . Is this just a (temporary) bug or a real difference?

**Comment by OskarStark** [comment]
@xabbuh @javiereguiluz please decide or not for an include

@ThomasLandauer can you please update the PR title to sth meaningful? 🙏

**Comment by wouterj** [comment]
👍 on making this an include

File: email.rst (hunk 1 above)
Comment (OskarStark) [reply email.rst]: No need to copy the link? @javiereguiluz

Comment (javiereguiluz) [reply email.rst]: Yes. I added the link while merging. Thanks for reporting this.

**Pull request merged**